### PR TITLE
[Layout] Add ASLayoutElementStylability protocol

### PR DESCRIPTION
**Pull request by maicki**
Is adding convenience methods to form layout specifications like:

```objc
 [ASInsetLayoutSpec
  insetLayoutSpecWithInsets:UIEdgeInsetsMake(10, 10, 10, 10)
  child:[ASStackLayoutSpec
         stackLayoutSpecWithDirection:ASStackLayoutDirectionVertical
         spacing:10
         justifyContent:ASStackLayoutJustifyContentStart
         alignItems:ASStackLayoutAlignItemsStretch
         children:@[
          [titleTextNode styleWithBlock:^(ASLayoutElementStyle *style) {
            style.width = ASDimensionMake(100);
            style.flexGrow = 1.0;
          }],
          [messageTextNode styleWithBlock:^(ASLayoutElementStyle *style) {
            style.flexGrow = 1.0;
            style.flexShrink = 1.0;
          }]]]];
```

### CODE REVIEW

File: AsyncDisplayKit/Layout/ASLayoutElement.h [hunk 1]
```diff
@@ -363,4 +364,10 @@ extern NSString * const ASLayoutElementStyleLayoutPositionProperty;
 
 @end
 
+@protocol ASLayoutElementStylability
+
+- (id<ASLayoutElement>)styleWithBlock:(void (^)(__kindof ASLayoutElementStyle *style))styleBlock;
```
Comment (Adlai-Holler) [inline AsyncDisplayKit/Layout/ASLayoutElement.h]: I think the return type here can be `instancetype`

Comment (Adlai-Holler) [inline AsyncDisplayKit/Layout/ASLayoutElement.h]: What do you think of the name `styledWithBlock`? That emphasizes the fact that the method has a return value.

Comment (Adlai-Holler) [inline AsyncDisplayKit/Layout/ASLayoutElement.h]: Let's add the fancy `__attribute__((noescape))` before the block type.

**Comment by maicki** [comment]
@Adlai-Holler Comments addressed. In fact what do you think about `ASLayoutElementStylability` protocol name at all? I don't really know if this is a great name, but couldn't find a better name for  it yet ...

**Comment by Adlai-Holler** [comment]
@maicki Looks great! To the protocol name, it's not a great name but we can always change it (or even remove it) later with no great penalty so let's move forward with it for now.

**Pull request merged**